presto-tests/src/test/java/com/facebook/presto/server/TestServerInfoResource.java

```diff
@@ -43,22 +49,43 @@
 {
     private HttpClient client;
     private DistributedQueryRunner queryRunner;
+    private ThriftCodecManager thriftCodeManager;
+
+    private NodeState getNodeState(URI uri, boolean useThriftEncoding, Protocol thriftProtocol)
+    {
+        Request.Builder requestBuilder = useThriftEncoding ? ThriftRequestUtils.prepareThriftGet(thriftProtocol) : getJsonTransportBuilder(prepareGet());
+        Request request = requestBuilder.setHeader(PRESTO_USER, "user").setUri(uri).build();
+        if (useThriftEncoding) {
+            return client.execute(request, new ThriftResponseHandler<>(thriftCodeManager.getCodec(NodeState.class))).getValue();
+        }
+        else {
+            requestBuilder = getJsonTransportBuilder(prepareGet());
+            return client.execute(request, createJsonResponseHandler(jsonCodec(NodeState.class)));
+        }
+    }
+
+    @DataProvider
+    public Object[][] thriftEncodingToggle()
```

Comment:
Let us move this method should be  after `@BeforeClass` method , since that is the convention. Please check other places we are using dataproviders.